nextcord/ext/commands/bot.py

```diff
@@ -149,9 +162,80 @@ class MissingMessageContentIntentWarning(UserWarning):
     pass
 
 
+_NonCallablePrefix = Union[str, Sequence[str]]
+
+
 class BotBase(GroupMixin):
-    def __init__(self, command_prefix=MISSING, help_command=_default, description=None, **options):
-        super().__init__(**options)
+    def __init__(
+        self,
+        command_prefix: Union[
+            _NonCallablePrefix,
+            Callable[[Self, Message], Union[Awaitable[_NonCallablePrefix], _NonCallablePrefix]],
+        ] = tuple(),
+        help_command: Optional[HelpCommand] = MISSING,
+        description: Optional[str] = None,
+        *,
+        max_messages: Optional[int] = 1000,
+        connector: Optional[aiohttp.BaseConnector] = None,
+        proxy: Optional[str] = None,
+        proxy_auth: Optional[aiohttp.BasicAuth] = None,
+        shard_id: Optional[int] = None,
+        shard_count: Optional[int] = None,
+        application_id: Optional[int] = None,
+        intents: nextcord.Intents = nextcord.Intents.default(),
+        member_cache_flags: MemberCacheFlags = MISSING,
+        chunk_guilds_at_startup: bool = MISSING,
+        status: Optional[Status] = None,
+        activity: Optional[BaseActivity] = None,
+        allowed_mentions: Optional[AllowedMentions] = None,
+        heartbeat_timeout: float = 60.0,
+        guild_ready_timeout: float = 2.0,
+        assume_unsync_clock: bool = True,
+        enable_debug_events: bool = False,
+        loop: Optional[asyncio.AbstractEventLoop] = None,
+        lazy_load_commands: bool = True,
+        rollout_associate_known: bool = True,
+        rollout_delete_unknown: bool = True,
+        rollout_register_new: bool = True,
+        rollout_update_known: bool = True,
+        rollout_all_guilds: bool = False,
+        owner_id: Optional[int] = None,
+        owner_ids: Iterable[int] = set(),
+        strip_after_prefix: bool = False,
+        case_insensitive: bool = False,
+    ):
+        nextcord.Client.__init__(
```

Comment:
This could be `nextcord.Client` or `nextcord.AutoShardedClient`.

I think this is what broke `AutoShardedBot`s